madsim/src/sim/task/mod.rs

```diff
@@ -266,18 +266,25 @@ impl Executor {
                 // future has been dropped
                 continue;
             };
-            if info.cancelled.load(Ordering::Relaxed) || info.node.killed.load(Ordering::Relaxed) {
+            let work = if info.cancelled.load(Ordering::Relaxed)
+                || info.node.killed.load(Ordering::Relaxed)
+            {
                 // cancelled task or killed node: drop the future
-                continue;
+                drop
```

Comment:
Ohhhh I see. The `continue` triggered us to directly proceed to the end of the loop, and call the destructor of runnable implicitly. Then it calls the drop for the struct, which spawned the thread.

Seems acceptable to me to drop it.